extensions/amp-brightcove/0.1/amp-brightcove.js

```diff
@@ -123,17 +128,32 @@ class AmpBrightcove extends AMP.BaseElement {
     ));
   }
 
-  /** @override */
-  layoutCallback() {
-    const iframe = createFrameFor(this, this.getIframeSrc_());
+  /**
+   * @return {!Promise}
+   */
+  getConsentState() {
+    const consentPolicyId = super.getConsentPolicy();
+    return consentPolicyId
+      ? getConsentPolicyState(this.element, consentPolicyId)
+      : Promise.resolve(null);
+  }
 
-    this.iframe_ = iframe;
+  /**
+   * @override
+   * @return {!Promise}
+   */
+  layoutCallback() {
+    this.getIframeSrc_().then((src) => {
```

Comment:
`return this.getIframeSrc_()....`

Reply:
@mister-ben If I'm not mistaken, since layoutCallback returns a promise, I think this whole promise should be returned?

i.e. 
```
layoutCallback() {
  return this.getIframeSrc_().then(....)
}
```

Reply:
the tests time out with that change

Reply:
Existing tests or new ones?

Reply:
Existing tests. All of the tests expect the one that intentionally fails to get as far as this point.